helloworld/src/main/scala/com/salesforce/hw/OpBostonSimple.scala

```diff
@@ -0,0 +1,165 @@
+    // RAW FEATURE DEFINITIONS
+    /////////////////////////////////////////////////////////////////////////////////
+
+    // Define features using the OP types based on the data
+    val rowId = FeatureBuilder.Integral[BostonHouse].extract(_.rowId.toIntegral).asPredictor
+    val crim = FeatureBuilder.RealNN[BostonHouse].extract(_.crim.toRealNN).asPredictor
+    val zn = FeatureBuilder.RealNN[BostonHouse].extract(_.zn.toRealNN).asPredictor
+    val indus = FeatureBuilder.RealNN[BostonHouse].extract(_.indus.toRealNN).asPredictor
+    val chas = FeatureBuilder.PickList[BostonHouse].extract(x => Option(x.chas).toPickList).asPredictor
+    val nox = FeatureBuilder.RealNN[BostonHouse].extract(_.nox.toRealNN).asPredictor
+    val rm = FeatureBuilder.RealNN[BostonHouse].extract(_.rm.toRealNN).asPredictor
+    val age = FeatureBuilder.RealNN[BostonHouse].extract(_.age.toRealNN).asPredictor
+    val dis = FeatureBuilder.RealNN[BostonHouse].extract(_.dis.toRealNN).asPredictor
+    val rad = FeatureBuilder.Integral[BostonHouse].extract(_.rad.toIntegral).asPredictor
+    val tax = FeatureBuilder.RealNN[BostonHouse].extract(_.tax.toRealNN).asPredictor
+    val ptratio = FeatureBuilder.RealNN[BostonHouse].extract(_.ptratio.toRealNN).asPredictor
+    val b = FeatureBuilder.RealNN[BostonHouse].extract(_.b.toRealNN).asPredictor
+    val lstat = FeatureBuilder.RealNN[BostonHouse].extract(_.lstat.toRealNN).asPredictor
+    val medv = FeatureBuilder.RealNN[BostonHouse].extract(_.medv.toRealNN).asResponse
+
+
+    // Define a feature of type vector containing all the predictors you'd like to use
+    val features = Seq(crim, zn, indus, chas, nox, rm, age, dis, rad, tax, ptratio, b, lstat).transmogrify()
+
+    val labels = medv
+
+    val checkedFeatures = labels.sanityCheck(features, removeBadFeatures = true)
+
+    ////////////////////////////////////////////////////////////////////////////////
+    // WORKFLOW DEFINITION
+    /////////////////////////////////////////////////////////////////////////////////
+
+    // Define the model we want to use (here a simple linear regression) and get the resulting output
+
+    val prediction = RegressionModelSelector
+      .withTrainValidationSplit(
+        modelTypesToUse = Seq(OpLinearRegression))
+      .setInput(labels, checkedFeatures).getOutput()
+
+    val evaluator = Evaluators.Regression().setLabelCol(labels).setPredictionCol(prediction)
+
+    ////////////////////////////////////////////////////////////////////////////////
+    // WORKFLOW
+    /////////////////////////////////////////////////////////////////////////////////
+
+    val dataReader = DataReaders.Simple.csvCase[BostonHouse](path = Option(csvFilePath), key = _.rowId.toString())
+
+    val workflow = new OpWorkflow().setResultFeatures(prediction, labels).setReader(dataReader)
+
+    val model = workflow.train()
+
+    println(s"Model summary:\n${model.summaryPretty()}")
+
```

Comment:
Actually, since it's so commonly used - can you add a piece here that extracts out the topN features by contribution from the ModelInsights (to all of the examples)?

Reply:
I can, but would it be redundant if this information is already in the model summary?